oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoFullGcNodeBinSumBsonSize.java

```diff
@@ -0,0 +1,137 @@
+ */
+class MongoFullGcNodeBinSumBsonSize implements FullGcNodeBin {
+    private static final Logger LOG = LoggerFactory.getLogger(MongoFullGcNodeBinSumBsonSize.class);
+
+    private final MongoFullGcNodeBin delegate;
+    private final MongoDocumentStore store;
+
+    public MongoFullGcNodeBinSumBsonSize(MongoFullGcNodeBin delegate) {
+        this.delegate = delegate;
+        this.store = delegate.getMongoDocumentStore();
+    }
+
+    @Override
+    public void setEnabled(boolean value) {
+        delegate.setEnabled(value);
+    }
+
+    /**
+     * Remove the documents from the collection and sum the the bson size of the removed properties     
+     * @param updateOpList the list of the documents to be removed
+     * @return the list of the documents removed
+     */ 
+    @Override
+    public List<NodeDocument> findAndUpdate(List<UpdateOp> updateOpList) {
+        List<String> ids = updateOpList.stream().map(UpdateOp::getId).collect(Collectors.toList());
+        //get the total bson size before the update
+        long initialBsonSize = getBsonSize(ids);
+        LOG.debug("bson size before update: {}", initialBsonSize);
+        //remove garbage properties
+        List<NodeDocument> updated = delegate.findAndUpdate(updateOpList);
+        if (!updated.isEmpty()) {
+            //calculate the diff of the bson size after update
+            long afterUpdateBsonSize = getBsonSize(ids);
+            LOG.debug("bson size after update: {}", afterUpdateBsonSize);
+            if (initialBsonSize > 0 && afterUpdateBsonSize > 0) {
+                //sum up the removed bson size
+                addBsonSize(initialBsonSize - afterUpdateBsonSize);
+            }
+        }
+        return updated;
+    }
+    /**
+     * Remove the documents from the collection and sum the their bson size      
+     * @param orphanOrDeletedRemovalMap the map of the documents to be removed
+     * @return the number of documents removed
+     */
+    @Override
+    public int remove(Map<String, Long> orphanOrDeletedRemovalMap) {
+        //get the total bson size before the update
+        long bsonSize = getBsonSize(new ArrayList<>(orphanOrDeletedRemovalMap.keySet()));
+        LOG.debug("bson size before remove: {}", bsonSize);
+        //remove garbage documents
+        int removed = delegate.remove(orphanOrDeletedRemovalMap);
+        //sum up the removed bson size
+        if (removed > 0 && bsonSize > 0) {
+            addBsonSize(bsonSize);
+        }
+        return removed;
+    }
+
+    private void addBsonSize(long bsonSize) {
+        if (bsonSize <= 0) {
+            LOG.warn("bson size {} is not positive", bsonSize);
+            return;
+        }
+        //sum the bson size with the value from fullGcBsonSize document in the SETTINGS collection
+        Bson query = eq(ID, "versionGC");
+        Bson update = Updates.inc("fullGcRemovedTotalBsonSize", bsonSize);
```

Comment:
please create a constant for this variable inside `VersionGarbageCollector` class along with other full gc constants